nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
```

Comment:
I don't disagree that it would be faster but I wasn't clear on why you think we need to optimize these containment checks and change the structure of the code.  Although your code is more compact I think it's less readable but I agree it would also be a bit faster to use a set instead of a list here (there are maybe a dozen elements here so I don't think it would really make a huge difference).  I think it's cleaner to leave the for loop with nested if's and you can explicitly see that the execution won't get to 216 unless all elements of subjclass (or their expansions) are recognized.  I agree the two loops can be pulled out to a helper function if you think that would be helpful.